tests/lib/rules/prefer-template.js

```diff
@@ -34,7 +34,8 @@ ruleTester.run("prefer-template", rule, {
 
         // https://github.com/eslint/eslint/issues/3507
         "var foo = `foo` + `bar` + \"hoge\";",
-        "var foo = `foo` +\n    `bar` +\n    \"hoge\";"
+        "var foo = `foo` +\n    `bar` +\n    \"hoge\";",
+        "var foo = `foo` + `\\0`;"
```

Comment:
I think these "valid" tests aren't working correctly -- the strings in these tests are already template literals, so the rule isn't reporting them because there would be nothing to adjust anyway.

From testing out this branch locally, I think the string `\0` is still preventing an autofix.